nbclient/client.py

```diff
@@ -330,22 +380,47 @@ def start_kernel_manager(self):
         if self.km.ipykernel and self.ipython_hist_file:
             self.extra_arguments += ['--HistoryManager.hist_file={}'.format(self.ipython_hist_file)]
 
-        self.km.start_kernel(extra_arguments=self.extra_arguments, **kwargs)
+        # Support AsyncKernelManager
+        if inspect.iscoroutinefunction(self.km.start_kernel):
```

Comment:
We could also use `asyncio.iscoroutinefunction`, but I guess this is fine.

Reply:
Based on https://stackoverflow.com/questions/36076619/test-if-function-or-method-is-normal-or-asynchronous the `asyncio` is probably a better choice to cover more cases. I'll change em over.